beancount_web/templates/_layout.html

```diff
@@ -56,8 +56,8 @@ <h1 class="site-name">{{ title or 'Beancount Web' }}</h1>
             {% for title_, menuitems in navigation_bar %}
                 {% if title_ %}<h3>{{ title_ }}</h3>{% endif %}
                 <ul class="navigation">
-                {% for href, id, caption in menuitems %}
-                    <li{% if id == active_page %} class="selected"{% endif %}><a href="{{ href|e }}">{{ caption|e }}{% if id == 'errors' and errors|length > 0 %}<span>{{ errors|length }}</span>{% endif %}</a></li>
+                {% for id, caption in menuitems %}
+                    <li{% if id == active_page %} class="selected"{% endif %}><a href="{{ url_for('report', report_name=id) }}">{{ caption|e }}{% if id == 'errors' and errors|length > 0 %}<span>{{ errors|length }}</span>{% endif %}</a></li>
```

Comment:
The url route for report is generic enough that it would match any url of the form `/some_string/`. That's why the more specific url routes come first, but `url_for` doesn't know that and happily generates the url for `source` :D